sed/core.py

```diff
@@ -76,17 +97,48 @@ def load(self, data: pd.DataFrame) -> None:
         """
         self._dataframe = pd.DataFrame(data)
 
+    def gen_jittered_df(self, cols: Sequence[str] = None) -> None:
```

Comment:
Is there a strong reason for creating a second dataframe, or could we not add these as columns in the same dataframe and then at binning select the columns with '_jittered' in the name?

Also, it might be that when creating new columns with, for example, calibrations, based on one columns, and later, but before computation jittering the original column in place, would induce the jitter to be computed before the computation.
This would allow us to apply the jitter function optionally just before binning.

Reply:
> Is there a strong reason for creating a second dataframe, or could we not add these as columns in the same dataframe and then at binning select the columns with '_jittered' in the name?

I had thought about this. This option has two disadvantages:
1. The binning of the "unjittered" case also has the computational overhead of the "jittered" case, as the jittering operations are part of the dataframe tasks (in practice, this limitation is somewhat irrellevant, as most of the time jittering will be used)
2. During binning, the names of the jittered and not-jittered dataframe columnes are different, so the user must be aware of this, and provide the corresponding name. Off course we could also rewrite the axes-names instead of the dataframe, but I found this solution cleaner. We could also drop it entirely, and only leave the jittering option as part of the dataframe operation queue, and remove it from the binning function.

> 
> Also, it might be that when creating new columns with, for example, calibrations, based on one columns, and later, but before computation jittering the original column in place, would induce the jitter to be computed before the computation. This would allow us to apply the jitter function optionally just before binning.

I also thought about this. This has the disadvantage, that this option then changes the dataframe in an irreversible manner. Afterwards, you cannot bin without jitter anymore. The binning function should not change the dataframe.

Reply:
I see your point, but still am not convinced:
1. the overhead of the jittered case applies only if you "use" the jittered column. If you don't, this is never computed as binning is computed on the subset of columns to be binned.
2. The user doesn't  need to specify the names. If we want to use the jittered ones (with jittered=True) then we only need to check for each binning axis called "name", if there is a column called "name_jittered" and pass that one instead.
3. dropping it entirely would be the cleanest, but as you mention, we don't want to modify the df in the process. However, we never overwrite the data from disk, so nothing stops us from reloading the data, or even keeping a pointer to the "clean" data and "resetting" a column if required.

Reply:
> * the overhead of the jittered case applies only if you "use" the jittered column. If you don't, this is never computed as binning is computed on the subset of columns to be binned.

Yes it is, I tried it out. You can do benchmarking yourself, or I can post code here if you want...
> 
> * The user doesn't  need to specify the names. If we want to use the jittered ones (with jittered=True) then we only need to check for each binning axis called "name", if there is a column called "name_jittered" and pass that one instead.

True, but I have the impression it would make the code a lot "messier", and in this case you also have to apply the jitter first. I find the solution with two dataframes cleaner. My idea was to apply all following df operations transparently on both dfs.
> 
> * dropping it entirely would be the cleanest, but as you mention, we don't want to modify the df in the process. However, we never overwrite the data from disk, so nothing stops us from reloading the data, or even keeping a pointer to the "clean" data and "resetting" a column if required.

Well, if we decide to drop the non-jittered dataframe at the point of applying jitter _within the dataframe operations chain_ then this is where alteration of the dataframe is expected, and this would be fine. But then we should remove the "jittered" option from the binning function. This would be the cleanest version, but for non-jittered binning one would need to start over. Actually, I prefer that one, now that I think more about it.

Reply:
We could leave the jittering on the binned coordinates in the binning function as an (undocumented) **kwrds argument option also in the code, or would this be confusing?